diff --git a/services/core/java/com/android/server/pm/PackageManagerService.java b/services/core/java/com/android/server/pm/PackageManagerService.java
index 2e63c6c4d4e3..cb809daf7b0e 100644
--- a/services/core/java/com/android/server/pm/PackageManagerService.java
+++ b/services/core/java/com/android/server/pm/PackageManagerService.java
@@ -20141,9 +20141,11 @@ Slog.v(TAG, ":: stepped forward, applying functor at tag " + parser.getName());
                 assertPackageKnown(volumeUuid, pkg.packageName);
 
             } catch (PackageParserException | PackageManagerException e) {
-                logCriticalInfo(Log.WARN, "Destroying " + file + " due to: " + e);
-                synchronized (mInstallLock) {
-                    removeCodePathLI(file);
+		if (!file.getAbsolutePath().equals("/data/app/mcRegistry"))  {
+	        		logCriticalInfo(Log.WARN, "Destroying " + file + " due to: " + e);
+	        		synchronized (mInstallLock) {
+	        			removeCodePathLI(file);
+                	}
                 }
             }
         }
