diff --git a/src/com/android/managedprovisioning/task/DownloadPackageTask.java b/src/com/android/managedprovisioning/task/DownloadPackageTask.java
index f35f98e7..caa3f731 100644
--- a/src/com/android/managedprovisioning/task/DownloadPackageTask.java
+++ b/src/com/android/managedprovisioning/task/DownloadPackageTask.java
@@ -15,8 +15,6 @@
  */
 package com.android.managedprovisioning.task;
 
-import static android.provider.Settings.Secure.MANAGED_PROVISIONING_DPC_DOWNLOADED;
-
 import android.app.DownloadManager;
 import android.app.DownloadManager.Query;
 import android.app.DownloadManager.Request;
@@ -31,7 +29,6 @@ import android.content.pm.Signature;
 import android.database.Cursor;
 import android.net.Uri;
 import android.text.TextUtils;
-import android.provider.Settings;
 
 import com.android.managedprovisioning.NetworkMonitor;
 import com.android.managedprovisioning.ProvisionLogger;
@@ -103,9 +100,6 @@ public class DownloadPackageTask {
                     + " the package");
             mCallback.onError(ERROR_OTHER);
         }
-
-        setDpcDownloadedSetting(mContext);
-
         mReceiver = createDownloadReceiver();
         mContext.registerReceiver(mReceiver,
                 new IntentFilter(DownloadManager.ACTION_DOWNLOAD_COMPLETE));
@@ -139,16 +133,6 @@ public class DownloadPackageTask {
         }
     }
 
-    /**
-     * Set MANAGED_PROVISIONING_DPC_DOWNLOADED to 1, which will prevent restarting setup-wizard.
-     *
-     * <p>See b/132261064.
-     */
-    private static void setDpcDownloadedSetting(Context context) {
-        Settings.Secure.putInt(
-                context.getContentResolver(), MANAGED_PROVISIONING_DPC_DOWNLOADED, 1);
-    }
-
     private BroadcastReceiver createDownloadReceiver() {
         return new BroadcastReceiver() {
             /**
